doc/specs/#607 - Commandline Arguments for the Windows Terminal.md

```diff
@@ -0,0 +1,534 @@
+wt cmd.exe
+
+# Runs the user's "Windows Powershell" profile in a new tab
+wt new-tab --profile "Windows Powershell"
+wt --profile "Windows Powershell"
+
+# run "my-commandline.exe with some args" in a new tab
+wt new-tab my-commandline.exe with some args
+wt my-commandline.exe with some args
+
+# run "my-commandline.exe with some args and a ; literal semicolon" in a new
+#  tab, and in another tab, run "another.exe running in a second tab"
+wt my-commandline.exe with some args and a \; literal semicolon
+
+# Start the default profile in directory "c:/Users/Foo/dev/MyProject" (user story 8)
+wt new-tab --startingDirectory "c:/Users/Foo/dev/MyProject"
+wt --startingDirectory "c:/Users/Foo/dev/MyProject"
+wt -d "c:/Users/Foo/dev/MyProject"
+
+# Runs the user's "Windows Powershell" profile in a new tab in directory
+#  "c:/Users/Foo/dev/MyProject" (user story 2, 8)
+wt new-tab --profile "Windows Powershell" --startingDirectory "c:/Users/Foo/dev/MyProject"
+wt --profile "Windows Powershell" --startingDirectory "c:/Users/Foo/dev/MyProject"
+wt -p "Windows Powershell" -d "c:/Users/Foo/dev/MyProject"
+
+```
+
+## Implementation Details
+
+TODO: I'm leaving this empty for the time being. I want to get some feedback on
+the proposal in this state, before starting work on implementation details. This
+style of arguments might be controversial, so I want to make sure we settle on a
+syntax before I get too far into the details of parsing and passing these args
+around.
+
+## Capabilities
+
+### Accessibility
+
+As a commandline feature, the accessibility of this feature will largely be tied
+to the ability of the commandline environment to expose accessibility
+notifications. Both `conhost.exe` and the Windows Terminal already support
+basica accessibility patterns, so users using this feature from either of those
+terminals will be reliant upon their accessibility implementations.
+
+### Security
+
+As we'll be parsing user input, that's always subject to worries about buffer
+length, input values, etc. Fortunately, most of this should be handled for us by
+the operating system, and passed to us as a commandline via `winMain` and
+`CommandLineToArgvW`. We should still take extra care in parsing these args.
+
+### Reliability
+
+This change should not have any particular reliability concerns.
+
+### Compatibility
+
+This change should not regress any existing behaviors.
+
+### Performance, Power, and Efficiency
+
+This change should not particularily impact startup time or any of these other categories.
+
+## Potential Issues
+
+#### Commandline escaping
+
+Escaping commandlines is notoriously tricky to do correctly. Since we're using
+`;` to delimit commands, which might want to also use `;` in the commandline
+itself, we'll use `\;` as an escaped `;` within the commandline. This is an area
+we've been caught in before, so extensive testing will be necessary to make sure
+this works as expected.

```

Comment:
I think you should call out that you seem to expressly be avoiding somehow delimiting the arguments on the command line to make life easier for folks who are launching something with arguments. Mismatching 500 escaping mechanisms is the bane of our existence for things like "launching an executable through cmd through powershell through tshell to a device". At least, that was my understanding as to why you didn't have a `--arguments "foo arguments"` option after the command line for a launch.

Reply:
**PowerShell** uses&nbsp;`;` as&nbsp;command&nbsp;separators, so:
```pwsh
wt new-tab cmd.exe ; split-pane -P 30 wsl.exe
```

Means:
1. Run `wt`(`.exe`) with `['new-tab', 'cmd.exe']` as the arguments.
2. Run `split-pane`(`.exe`) with `['-P', '30', 'wsl.exe']` as the arguments.
	- This&nbsp;will&nbsp;throw an&nbsp;exception, unless&nbsp;the&nbsp;user has&nbsp;a&nbsp;`split‑pane` command on&nbsp;their&nbsp;`PATH` or&nbsp;as&nbsp;a&nbsp;**PowerShell** cmdlet&nbsp;

Reply:
@ExE-Boss thanks for the breakdown. Clearly, I'm more of a `cmd` guy 😅.

I've added more details in this section to elaborate upon this point. Let's use this thread to discuss. Technically, we could ask the user to escape our commandline in powershell, like ```wt new-tab `; new-pane commandline \`; with \`; semicolons``` or ```wt new-tab ';' new-pane "commandline \; with \; semicolons"```. However, if the party line is "use powershell", does it really make sense for us to add a commandline that _fundamentally_ needs to be escaped when running in powershell? What other command separators could we use?

Reply:
`--`?